### ISSUE TITLE
Ignore startup timeouts if the debugger is attached

### ISSUE THREAD

**Issue by Tratcher** [aspnet/IISIntegration#71]
HttpPlatformHandler has a startup time limit for how long it will wait for the back-end process to accept requests. Then it kills the process and tries again up to 10 times. However, if we want to debug anything at startup we usually exceed these limits so we have to override them with `startupTimeLimit="3600"`.

Proposal: HttpPlatformHandler should ignore this time limit while the debugger is attached.

Attached to what process? The back-end process? IIS Express? Both?

**Comment by Tratcher** [comment]
Note the startup process for VS F5 is to:
1) launch IIS express
2) Make a DEBUG request to the web site so get it started
3) Attach to the process

@BillHiebert

**Comment by Tratcher** [comment]
Rumor has it this was fixed back in the 10/21 release. Needs verification.

**Comment by Tratcher** [comment]
Fixed in the next release of HttpPlatformHandler (v1.3?).

However, we've got some tooling work to do on our side. When we switch to the .NET CLI we'll loose our DNX MITM that waits for the debbugger to attach before running user code. Tentative theory is for VS to replace this with it's own debug launcher that starts the back-end process suspended, waits for the debugger to attach, and then resumes it. The intermediate process can then exit because HttpPlatformHandler only monitors the process that is listening on the socket. @BillHiebert
